core/src/main/java/org/apache/accumulo/core/util/tables/TableDiskUsage.java

```diff
@@ -0,0 +1,180 @@
+    // get the internal id for this table
+    int internalId = internalIds.get(tableId);
+
+    // Initialize a bitset for tables (internal IDs) that reference this file
+    Integer[] tables = tableFiles.get(file);
+    if (tables == null) {
+      tables = new Integer[internalIds.size()];
+      for (int i = 0; i < tables.length; i++) {
+        tables[i] = 0;
+      }
+      tableFiles.put(file, tables);
+    }
+
+    // Update the bitset to track that this table has seen this file
+    tables[internalId] = 1;
+  }
+
+  protected void addFileSize(String file, long size) {
+    fileSizes.put(file, size);
+  }
+
+  protected Map<List<TableId>,Long> calculateSharedUsage() {
+    // Bitset of tables that contain a file and total usage by all files that share that usage
+    Map<List<Integer>,Long> usage = new HashMap<>();
+
+    if (log.isTraceEnabled()) {
+      log.trace("fileSizes {}", fileSizes);
+    }
+    // For each file w/ referenced-table bitset
+    for (Entry<String,Integer[]> entry : tableFiles.entrySet()) {
+      if (log.isTraceEnabled()) {
+        log.trace("file {} table bitset {}", entry.getKey(), Arrays.toString(entry.getValue()));
+      }
+      List<Integer> key = Arrays.asList(entry.getValue());
+      Long size = fileSizes.get(entry.getKey());
+
+      Long tablesUsage = usage.getOrDefault(key, 0L);
+      tablesUsage += size;
+      usage.put(key, tablesUsage);
+    }
+
+    final Map<List<TableId>,Long> externalUsage = new HashMap<>();
+
+    for (Entry<List<Integer>,Long> entry : usage.entrySet()) {
+      List<TableId> externalKey = new ArrayList<>();
+      List<Integer> key = entry.getKey();
+      // table bitset
+      for (int i = 0; i < key.size(); i++)
+        if (key.get(i) != 0) {
+          // Convert by internal id to the table id
+          externalKey.add(externalIds.get(i));
+        }
+
+      // list of table ids and size of files shared across the tables
+      externalUsage.put(externalKey, entry.getValue());
+    }
+
+    // mapping of all enumerations of files being referenced by tables and total size of files who
+    // share the same reference
+    return externalUsage;
+  }
+
+  protected static SortedMap<SortedSet<String>,Long> buildSharedUsageMap(final TableDiskUsage tdu,
+      final ClientContext clientContext, final Set<TableId> emptyTableIds) {
+    final Map<TableId,String> reverseTableIdMap = clientContext.getTableIdToNameMap();
+
+    final SortedMap<SortedSet<String>,Long> usage = new TreeMap<>((o1, o2) -> {
+      int len1 = o1.size();
+      int len2 = o2.size();
+
+      int min = Math.min(len1, len2);
+      Iterator<String> iter1 = o1.iterator();
+      Iterator<String> iter2 = o2.iterator();
+
+      int count = 0;
+      while (count < min) {
+        String s1 = iter1.next();
+        String s2 = iter2.next();
+
+        int cmp = s1.compareTo(s2);
+        if (cmp != 0) {
+          return cmp;
+        }
+
+        count++;
+      }
+
+      return len1 - len2;
+    });
```

Comment:
This comparator seems to be non-trivial, but has no comments explaining what it's trying to do

Reply:
The comparator already existed previously, I didn't change anything with it, I just moved it. I can look into more at what it's doing but it wasn't something I created.

https://github.com/apache/accumulo/blob/87402dfcd8d828850ca2e4b01e8811cf506e7d4e/server/base/src/main/java/org/apache/accumulo/server/util/TableDiskUsage.java#L231-L257